src/NodeServices/MessageEvents.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace DynamoServices
+{
+    public class LogWarningMessageEventArgs : EventArgs
+    {
+        public string message { get; internal set; }
+
+        public LogWarningMessageEventArgs(string msg)
+        {
+            message = msg;
+        }
+    }
+
+    public delegate void LogWarningMessageEventHandler(LogWarningMessageEventArgs args);
```

Comment:
Note that the other message logging events can be added in addition to just warning message.
The reason we have to be explicit about warning message is because a warning will also highlight the node while displaying the message.